gdx/src/com/badlogic/gdx/scenes/scene2d/utils/TiledDrawable.java

```diff
@@ -43,50 +46,226 @@ public void draw (Batch batch, float x, float y, float width, float height) {
+
+		final Texture texture = textureRegion.getTexture();
+		final float textureWidth = texture.getWidth() * scale;
+		final float textureHeight = texture.getHeight() * scale;
+		final float u = textureRegion.getU();
+		final float v = textureRegion.getV();
+		final float u2 = textureRegion.getU2();
+		final float v2 = textureRegion.getV2();
+
+		int fullX = (int)(width / regionWidth);
+		final float leftPartialWidth;
+		final float rightPartialWidth;
+		if (Align.isLeft(align)) {
+			leftPartialWidth = 0f;
+			rightPartialWidth = width - (regionWidth * fullX);
+		} else if (Align.isRight(align)) {
+			leftPartialWidth = width - (regionWidth * fullX);
+			rightPartialWidth = 0f;
+		} else {
+			if (fullX != 0) {
+				fullX = fullX % 2 == 1 ? fullX : fullX - 1;
+				final float leftRight = 0.5f * (width - (regionWidth * fullX));
+				leftPartialWidth = leftRight;
+				rightPartialWidth = leftRight;
+			} else {
+				leftPartialWidth = 0f;
+				rightPartialWidth = 0f;
+			}
+		}
+		int fullY = (int)(height / regionHeight);
+		final float topPartialHeight;
+		final float bottomPartialHeight;
+		if (Align.isTop(align)) {
+			topPartialHeight = 0f;
+			bottomPartialHeight = height - (regionHeight * fullY);
+		} else if (Align.isBottom(align)) {
+			topPartialHeight = height - (regionHeight * fullY);
+			bottomPartialHeight = 0f;
+		} else {
+			if (fullY != 0) {
+				fullY = fullY % 2 == 1 ? fullY : fullY - 1;
+				final float topBottom = 0.5f * (height - (regionHeight * fullY));
+				topPartialHeight = topBottom;
+				bottomPartialHeight = topBottom;
+			} else {
+				topPartialHeight = 0f;
+				bottomPartialHeight = 0f;
 			}
-			x += regionWidth;
 		}
-		Texture texture = region.getTexture();
-		float u = region.getU();
-		float v2 = region.getV2();
-		if (remainingX > 0) {
-			// Right edge.
-			float u2 = u + remainingX / (texture.getWidth() * scale);
-			float v = region.getV();
-			y = startY;
-			for (int ii = 0; ii < fullY; ii++) {
-				batch.draw(texture, x, y, remainingX, regionHeight, u, v2, u2, v);
-				y += regionHeight;
+
+		float drawX = x;
+		float drawY = y;
+
+		// Left edge
+		if (leftPartialWidth > 0f) {
+			final float leftEdgeU = u2 - (leftPartialWidth / textureWidth);
+
+			// Left bottom partial
+			if (bottomPartialHeight > 0f) {
+				final float leftBottomV = v + (bottomPartialHeight / textureHeight);
+				batch.draw(texture, drawX, drawY, leftPartialWidth, bottomPartialHeight, leftEdgeU, leftBottomV, u2, v);
+				drawY += bottomPartialHeight;
 			}
-			// Upper right corner.
-			if (remainingY > 0) {
-				v = v2 - remainingY / (texture.getHeight() * scale);
-				batch.draw(texture, x, y, remainingX, remainingY, u, v2, u2, v);
+
+			// Left center partials
+			if (fullY == 0 && Align.isCenterVertical(align)) {
+				final float vOffset = 0.5f * (v2 - v) * (1f - (height / regionHeight));
+				final float leftCenterV = v2 - vOffset;
+				final float leftCenterV2 = v + vOffset;
+				for (int i = 0; i < 1; i++) {
```

Comment:
This is either a bug (loop can only execute once) or the loop can be removed. This occurs 4 times in this file.

Reply:
It seems to behave correctly, so it looks like it's just an extraneous loop around code that should be kept.

Reply:
Good catch, thank you! I removed the redundant loops.

Reply:
Sorry for the delay. I still see the loops? (`for (int i = 0; i < 1; i++) {`) Maybe you didn't push?

Reply:
Sorry for that! I just pushed again. Most likely something didn't work properly as we are moving houses at the moment and I'm working from another laptop. Github also wanted me to confirm strange stuff for access this time...

Reply:
Thanks! Sorry for the delay. For some reason I don't get notifications about this PR, despite being subscribed to all events.